packages/application-to-register-webapp/src/routes/__tests__/land/legal-agreement-start-date-spec.js

```diff
@@ -0,0 +1,126 @@
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should add a legal agreement start date and back to referrer', async () => {
+      const h = {
+        view: jest.fn()
+      }
+      const redisMap = new Map()
+      redisMap.set(constants.redisKeys.LEGAL_AGREEMENT_START_DATE_KEY, '22-10-2022')
+      const request = {
+        yar: redisMap,
+        info: {
+          referrer: 'check-legal-agreement-details'
+        }
+      }
+      const legalAgreementParties = require('../../land/legal-agreement-start-date')
+      await legalAgreementParties.default[0].handler(request, h)
+      postOptions.payload['legalAgreementStartDate-day'] = '01'
+      postOptions.payload['legalAgreementStartDate-month'] = '01'
+      postOptions.payload['legalAgreementStartDate-year'] = '1971'
+
+      const response = await submitPostRequest(postOptions)
+      expect(response.statusCode).toBe(302)
+    })
+
+    it('should fail to add a legal agreement start date with empty dates', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = ''
+      postOptions.payload['legalAgreementStartDate-month'] = ''
+      postOptions.payload['legalAgreementStartDate-year'] = ''
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty day', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = ''
+      postOptions.payload['legalAgreementStartDate-month'] = '01'
+      postOptions.payload['legalAgreementStartDate-year'] = '2022'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty month', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = '01'
+      postOptions.payload['legalAgreementStartDate-month'] = ''
+      postOptions.payload['legalAgreementStartDate-year'] = '2022'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with empty year', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = '01'
+      postOptions.payload['legalAgreementStartDate-month'] = '01'
+      postOptions.payload['legalAgreementStartDate-year'] = ''
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
+    })
+
+    it('should fail to add a legal agreement start date with invalid dates', async () => {
+      postOptions.payload['legalAgreementStartDate-day'] = 'xx'
+      postOptions.payload['legalAgreementStartDate-month'] = 'zz'
+      postOptions.payload['legalAgreementStartDate-year'] = 'cc'
+
+      const response = await submitPostRequest(postOptions, 200)
+      expect(response.statusCode).toBe(200)
```

Comment:
Please can you add additional checks for expected errors. [For example](https://github.com/DEFRA/biodiversity-net-gain-service/blob/master/packages/application-to-register-webapp/src/routes/__tests__/land/add-landowners.spec.js).